src/__tests__/TicTacToe.test.tsx

```diff
@@ -12,4 +12,89 @@ describe('TicTacToe component', () => {
     const wrapper = shallow(<TicTacToe />);
     expect(wrapper.find('BoardSizeForm').exists()).toBe(true);
   });
+
+  it('will render the Board component if given invalid input', () => {
+    const wrapper = mount(<TicTacToe />);
+    const input = wrapper.find('#size-input');
+
+    (input.instance() as React.DetailedHTMLProps<
```

Comment:
Yeahh these are gross :) I would just use `as any` here; we know that the element is an input that has a value property; IMO it's not worth the verbosity to tell the test that, too.

Reply:
Alternatively, if the details of interacting with an element are messy and unavoidable like this (for example, selecting a value from a multiselect field), it's often nice to abstract these gross details into a helper method:

```jsx
fill('form input[type="text"]', { with: '3' });
```